### PR TITLE
[Serve] Raise error on creating detached serve instance with no long-running Ray cluster

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Previously, a user could call `serve.start(detached=True)` without having first called `ray.init(address="auto")`.  The controller would call `ray.init()`, which would start a new ray cluster that would be shutdown at the end of the script, killing the controller as well.  

This PR raises an error in this situation.  `serve.start(detached=True)` is only intended to be called when there is a long-running Ray cluster.  To be precise, if called when ray is not initialized, it will try to connect to a long-running cluster, and if this fails, it will raise an error.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -355,7 +356,11 @@ def start(detached: bool = False,
     """
     # Initialize ray if needed.
     if not ray.is_initialized():
-        ray.init()
+        if detached:
+            # Try to connect to long-running Ray cluster.  Error if we can't
```
Comment (edoakes) [inline python/ray/serve/api.py:360]:
```suggestion
            # Try to connect to existing Ray cluster, error if we can't.
```

**Review by edoakes — COMMENTED** [review]
Oh, I think I may have misunderstood the issue originally - I thought this was in `connect`. I don't think we want to change the behavior in `start` because you could do the following:
- Start local ray cluster in script, start detached instance on it.
- While first script is still running, connect to it with second script and modify detached instance.

**Comment by architkulkarni** [comment]
> Oh, I think I may have misunderstood the issue originally - I thought this was in `connect`.
Ah no, it's my fault, I accidentally wrote `connect` instead of `start` in my initial message to you.

Your use case makes sense, I hadn't thought of that.  Should we go ahead and close the PR?

**Comment by architkulkarni** [comment]
Or should we perhaps amend the comment in the docstring for `start` that reads `This is only relevant if connecting to a long-running Ray cluster (e.g., with address="auto").`?

**Pull request closed without merging**